src/App.tsx

```diff
@@ -10,38 +10,45 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from './app/store';
+import { authorActions } from './features/author/author';
+import { selectedPostActions } from './features/selectedPost/selectedPost';
+import { postsActions } from './features/posts/posts';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useDispatch();
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { posts, loaded, hasError } = useSelector(
+    (state: RootState) => state.posts,
+  );
+
+  const author = useSelector((state: RootState) => state.author);
+  const selectedPost = useSelector((state: RootState) => state.selectedPost);
```

Comment:
better to create useAppSelector and useAppDispatch hooks and remove types here

https://redux-toolkit.js.org/tutorials/typescript